drivers/mspi/mspi_dw.c

```diff
@@ -1223,6 +1263,37 @@ static int _api_xip_config(const struct device *dev,
 	return 0;
 }
 
+static int api_timing_config(const struct device *dev,
+			     const struct mspi_dev_id *dev_id,
+			     const uint32_t param_mask, void *cfg)
+{
+	struct mspi_dw_data *dev_data = dev->data;
+	struct mspi_dw_timing_cfg *config = cfg;
+	int rc, rc2;
+
+	if (param_mask & MSPI_DW_RX_TIMING_CFG) {
+		rc = pm_device_runtime_get(dev);
+		if (rc < 0) {
+			LOG_ERR("pm_device_runtime_get() failed: %d", rc);
+			return rc;
+		}
+
+		(void)k_sem_take(&dev_data->ctx_lock, K_FOREVER);
+		write_rx_sample_dly(dev, config->rx_sample_dly);
```

Comment:
I think it would be better to just store this value in `dev_data` here and write the register in `start_next_packet()`, together with the others that cannot be written when the controller is enabled (databook 1.03a says that's the case for this register, too).